src/main/java/io/confluent/idesidecar/restapi/models/graph/RealCCloudFetcher.java

```diff
@@ -534,4 +541,76 @@ protected CCloudOAuthContext.OrganizationDetails currentOrgDetails(String connec
     }
     return null;
   }
+
+  public Multi<FlinkComputePool> getFlinkComputePools(String connectionId, String envId) {
+    var headers = headersFor(connectionId);
+    String url = CONFLUENT_CLOUD_FLINK_COMPUTE_POOLS_URI.formatted(envId);
+    Log.infof("Fetching Flink compute pools from URL: %s with headers: %s", url, headers);
+    return listItems(headers, url, null, this::parseFlinkComputePoolsList)
+        .map(pool -> pool.withConnectionId(connectionId));
+  }
+
+  private PageOfResults<FlinkComputePool> parseFlinkComputePoolsList(String json, PaginationState state) {
+    return parseList(json, state, ListFlinkComputePoolsResponse.class);
+  }
+
+  private Multi<Map<String, String>> headersForAllConnections() {
+    // Implement logic to get headers for all connections
+    return Multi.createFrom().empty();
+  }
+
+  @RegisterForReflection
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private record ListFlinkComputePoolsResponse(
+      @JsonProperty(value = "api_version") String apiVersion,
+      String kind,
+      ListMetadata metadata,
+      @JsonProperty(value = "data", required = true) List<FlinkComputePoolResponse> data
+  ) implements ListResponse<FlinkComputePoolResponse, FlinkComputePool> {
+
+  }
+
+  @RegisterForReflection
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private record FlinkComputePoolResponse(
+      @JsonProperty(value = "api_version") String apiVersion,
+      String kind,
+      @JsonProperty(required = true) String id,
+      JsonNode metadata,
+      @JsonProperty(value = "spec")
+      io.confluent.idesidecar.restapi.models.graph.FlinkComputePoolSpec spec,
+      @JsonProperty(value = "status") io.confluent.idesidecar.restapi.models.graph.FlinkComputePoolStatus status,
+      @JsonProperty(value = "connection_id") String connectionId
```

Comment:
I'm not sure if CCloud returns this field. Do we need it?

Reply:
removed!